src/language_features/lean.rs

```diff
@@ -77,3 +77,76 @@ pub fn plain_goal(meta: EditorMeta, params: EditorPlainGoalParams, ctx: &mut Con
         },
     );
 }
+
+#[derive(Debug, Clone)]
+pub struct EditorPlainTermGoalParams {
+    pub position: KakounePosition,
+    pub buffer: String,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct PlainTermGoalParams {
+    pub text_document: TextDocumentIdentifier,
+    pub position: Position,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct PlainTermGoalResponse {
+    pub goal: String,
+    pub range: Range,
+}
+
+pub struct PlainTermGoalRequest {}
+
+impl Request for PlainTermGoalRequest {
+    type Params = PlainTermGoalParams;
+    type Result = Option<PlainTermGoalResponse>;
+    const METHOD: &'static str = "$/lean/plainTermGoal";
+}
+
+pub fn plain_term_goal(meta: EditorMeta, params: EditorPlainTermGoalParams, ctx: &mut Context) {
+    let req_params = ctx
+        .servers(&meta)
+        .map(|(server_id, server_settings)| {
+            (
+                server_id,
+                vec![PlainTermGoalParams {
+                    text_document: TextDocumentIdentifier {
+                        uri: Url::from_file_path(&meta.buffile).unwrap(),
+                    },
+                    position: get_lsp_position(
+                        server_settings,
+                        &meta.buffile,
+                        &params.position,
+                        ctx,
+                    )
+                    .unwrap()
+                }],
+            )
+        })
+        .collect();
+
+    ctx.call::<PlainTermGoalRequest, _>(
+        meta,
+        RequestParams::Each(req_params),
+        move |ctx: &mut Context, _meta, results| {
```

Comment:
rather than copying code, we should extract a function.
At least for the response handling (which is not just boilerplate),
but probably for the entire `plain_term_goal()`.

It will be a generic function; the two callers will pass PlainGoalRequest and PlainTermGoalRequest as concrete type parameters.

	fn plain_goal_impl<T: Request>(meta: EditorMeta, params: EditorPlainGoalParams, ctx: &mut Context) {
		...
		ctx.call::<T>(...);
	}